### PR TITLE
Tickets/dm 18844

### PR DESCRIPTION
**Pull request by fritzm**
Cleanups for various warnings generated by clang

### CODE REVIEW

File: core/modules/replica/ReplicationTask.h [hunk 1]
```diff
@@ -105,10 +105,6 @@ class ReplicationTask : public Task {
     /// on the Qserv synchronization requests.
     unsigned int _qservSyncTimeoutSec;
 
-    /// The number of seconds to wait in the end of each iteration loop before
-    /// to begin the new one.
-    unsigned int _replicationIntervalSec;
```
Comment (iagaponenko) [inline core/modules/replica/ReplicationTask.h]: 👍

File: core/modules/replica/HttpProcessor.h [hunk 2]
```diff
@@ -248,8 +248,6 @@ class HttpProcessor : public std::enable_shared_from_this<HttpProcessor> {
     // the reference counters to the pointed objects.
 
     HealthMonitorTask::Ptr const& _healthMonitorTask;
-    ReplicationTask::Ptr   const& _replicationTask;
-    DeleteWorkerTask::Ptr  const& _deleteWorkerTask;
```
Comment (iagaponenko) [inline core/modules/replica/HttpProcessor.h]: This is good for now. Though, I will need to put both back later when I will actually need these references. An idea behind having these references here was to allow REST API to suspend/resume those tasks if needed.

File: core/modules/replica/StatusRequest.h [hunk 3]
```diff
@@ -58,7 +58,8 @@ namespace replica {
 //   Customizations for specific request types require dedicated policies
 // ========================================================================
 
-struct StatusReplicationRequestPolicy {
+class StatusReplicationRequestPolicy {
```
Comment (iagaponenko) [inline core/modules/replica/StatusRequest.h]: The change is fine. I'm just curious about a reason for this change.

Comment (fritzm) [reply core/modules/replica/StatusRequest.h]: clang / llvm warns on inconsistent use of {{class}} vs {{struct}}, so we should pick one in each case.

And easier, I think, to just start using {{class}} everywhere, even for POD?  That way you don't have to think about rules of what is POD and what is not, have people interpret differently and argue, remember to change/update code when appropriate, etc.

File: core/modules/qana/TableInfo.h [hunk 4]
```diff
@@ -187,28 +187,28 @@ struct DirTableInfo : TableInfo {
     DirTableInfo(std::string const& db, std::string const& t, double o) :
         TableInfo(db, t, DIRECTOR), partitioningId(0), overlap(o) {}
 
-    virtual std::vector<ColumnRefConstPtr> const makeColumnRefs(
-        std::string const& tableAlias) const;
+    std::vector<ColumnRefConstPtr> const makeColumnRefs(
+        std::string const& tableAlias) const override;
 
-    virtual bool isEqPredAdmissible(TableInfo const& t,
+    bool isEqPredAdmissible(TableInfo const& t,
```
[the remark this answers is not in the document]
Comment (fritzm) [reply core/modules/qana/TableInfo.h]: thx; will fix...

File: core/modules/qana/DuplSelectExprPlugin.h [hunk 5]
```diff
@@ -81,15 +81,15 @@ class DuplSelectExprPlugin : public QueryPlugin {
      */
     static std::string const ERR_MSG;
 
-    virtual ~DuplSelectExprPlugin() {}
+    ~DuplSelectExprPlugin() override {}
```
[the remark this answers is not in the document]
Comment (fritzm) [reply core/modules/qana/DuplSelectExprPlugin.h]: Perhaps, consistently, throughout, on a future changelist.

**Pull request merged**